test/index.js

```diff
@@ -19,19 +19,25 @@ test('validate-npm-package-name', function (t) {
 
   // Scoped (npm 2+)
 
-  t.deepEqual(validate('@npm/thingy'), {validForNewPackages: true, validForOldPackages: true})
+  t.deepEqual(validate('@npm/thingy'), {
+    validForNewPackages: false,
+    validForOldPackages: true,
+    warnings: ['name cannot start with a special/capital character']
+  })
+
   t.deepEqual(validate('@npm-zors/money!time.js'), {
```

Comment:
We should also have a test case which checks for a special character in the middle instead of start.

Reply:
So, there is a condition that the package name should not contain `~)('!*` these characters.  Apart from these and `non-url-safe` (refer to the readme for this) characters, all the characters are allowed and we do have test cases to validate these conditions. 

I don' think we need another similar kind of test case. Let me know if I'm missing anything.

Reply:
Based on the new check flow would return from there as it starts with @ and the other condition that `!` is allowed in the middle of name would not get validated.